src/reanimated2/Hooks.js

```diff
@@ -433,51 +433,55 @@ export function useAnimatedGestureHandler(handlers, dependencies) {
   );
   initRef.current.savedDependencies = dependencies;
 
+  const handler = (event) => {
+    'worklet';
+    event = Platform.OS === 'web' ? event.nativeEvent : event;
```

Comment:
I'm pretty sure we should use `Platform.OS !== 'native'` here. This way we can potentially support RN for web, ios, and windows.

Reply:
by iOS here you mean macOS?

Reply:
what's the difference between checking for `native` and `web`? Aren't they just opposite(I mean `Platform.OS === web` == `Platform.OS !== native`)?

Reply:
> by iOS here you mean macOS?

Yeah, I meant macos.

> what's the difference between checking for native and web? Aren't they just opposite(I mean Platform.OS === web == Platform.OS !== native)?

I don't think that `Platform.OS === web` evaluates to true in windows or macOS version of RN, that's why I think we should go with "native or everything else" here.

Reply:
AFAIK we support 3 platforms for now: web, ios, and android. There haven't been any test for any other platforms that I am aware of. In this PR I attached specific behavior for the web. If you think we should handle windows and macOS then I think you should open a new issue for that as we check for OS versions in the same(or similar) way as here in multiple places in the code.

Reply:
Resolved internally, we came to the conclusion that we should focus on the web support for now, so although we should use the "not native" check here, this solution is not scalable and we'll have to address this problem in the future.

We can focus on implementation for other platforms if the community expresses interest.

Reply:
+1, also `'native'` won't work for such checks, closing both discussions.